### ISSUE TITLE
Choosing Feature, Enhancement for v0.9 iteration

### ISSUE THREAD

**Issue by jeevatkm** [go-aah/aah#107]
I would like to get aah users inputs before starting the `v0.9` iteration development. Please have a look roadmap and share your input. 

**I'm thinking of:**

**Enhancements** (planning to pick 2 items)
* Adding password encoders `scrypt`, `pbkdf2` for authentication
     * <s>Not decided yet for these two sha, md5</s>
* Context based logger enhancement and fields
* HTTP -> HTTPS redirects support

**Features** (planning to pick 2 items)
* TestSuite for integration test case and function test case  
* Websocket Module
* Server Sent Events (SSE) Module
* Validation Module
* CSRF

Also if anyone interested in contributing, please pick an area, besides my number. So that v0.9 release will have more things 4+.

Please post input via `comments` and `thumps up` for that comment.

**Comment by adelowo** [comment]
Hi,

*Enhancement*

- Password encoders : Can we get rid of md5 and bring in bcrypt ? I feel
md5 shouldn't be used for auth purposes or am I kissing something here ?
Maybe pbkdf2 can come in place of md5. Again, this are just my own thoughts.
- HTTP -> HTTPS redirects support

*Features *

- TestSuite for integration test case and function test case.

As per card, is this necessary we have this in since users can already
register middlewares ?. And aah supports the standard `http.Handler` -
right ? -, it should not be an issue importing one of the many awesome csrf
libraries out there.

**Comment by jeevatkm** [comment]
@adelowo 
* aah already has support for `bcrypt` since `v0.7` release
* Sure I will add `pbkdf2` into the list. I agree with you I thought of removing sha & md5 both but thought to get an input from aah users

Yes, aah support standard handler. Regarding csrf handler I will ping you via gitter later on. I did research around it.

**Comment by adelowo** [comment]
Oops, I missed that.. I haven't monitored the framework's progress in a long time.. Sure gitter works

**Comment by julienkosinski** [comment]
Thank you @jeevatkm for this, very nice to have this kind of discussion!

Yeah! I believe for security reason maybe it should only be possible to implement by our own SHA and MD5 if some people would like... Not userfriendly implemented.
bcrypt, scrypt and pbkdf2 are secure and useful for  to have I think! :)

For me I would really like and need as enhancement some OAuth implementation, but I guess that's kind of huge (except if you could find a good lib to do it properly).

**Comment by jeevatkm** [comment]
@julienkosinski You're welcome, actually it would help me to priorities the feature and enhancements.

* Per aah security moudle, aah user can register their own password encoder since `v0.7`. It just not documented yet 😄  So I will discount SHA and MD5.
* Could you please elaborate bit on OAuth part. Do you mean; you want to integrate with OAuth Services with aah application?
     * You can do it using Generic Auth capability. Yes I agree currently no ready to use pluggable option for OAuth and JWT (But I have provided tutorial using Generic Auth and JWT). Pluggable options, I'm planning to do it later, since I'm frying bigger fish 😃

**Comment by julienkosinski** [comment]
@jeevatkm 

- Ho, ok, great then!

- In fact I'm thinking of a server implementation of OAuth protocol to be able to expose a OAuth framework from aah. Client pluggable options could be nice too :).

**Comment by jeevatkm** [comment]
@julienkosinski Got it. It's server OAuth implementation you were taking about. Let me quickly check one thing and get back on this.

**Comment by jeevatkm** [comment]
@julienkosinski I have just verified my doubts on OAuth server.

* It is better to use OAuth Server libraries ([osin](https://github.com/RangelReale/osin), [go-oauth2](https://github.com/go-oauth2/oauth2)) of your choice with `aah`
* Use ready to deploy OAuth server like [hydra](https://github.com/ory/hydra), etc

Adding pluggable OAuth2 option in security module is helpful not a server library. 

I'm happy to assist you for your implementation, you can ping me anytime on gitter privately.

**Comment by julienkosinski** [comment]
Okay then, perfect, thank you very much! :D

**Comment by blagodus** [comment]
Enhancements:
- Adding password encoders scrypt, pbkdf2 for authentication
- HTTP -> HTTPS redirects support

Features:
- Validation Module
- CSRF

**Comment by jeevatkm** [comment]
@blagodus thank you for inputs.

**Comment by jeevatkm** [comment]
Thank you for everyone's inputs.
